### ISSUE TITLE
API idea.

### ISSUE THREAD

**Issue by stefanpenner** [stefanpenner/ember-state-services#28]
rather then inject +  cp

it would be nice to do.

```
state: stateFor('email', { key: 'email' })

// or

state: stateFor('email', { key: function(attr, state) { return // w/e you can think of})
```

in most cases we can likely auto-generate the state service itself, only making the state objects something the user needs to deal with.

cc @thoov

**Comment by thoov** [comment]
@stefanpenner I agree with auto-generating the state service or at least doing something behind the scenes so that it is hidden from the users point of view.

How would the user get `stateFor` in your example if they are not going to inject it?

**Comment by stefanpenner** [comment]
> How would the user get stateFor in your example if they are not going to inject it?

import

**Comment by thoov** [comment]
If you simply import stateFor how would it get access to the container?

I assume you are talking about something like this:


```js
// state-for.js
export default function(stateName, options) {
  



};

// my-cool-component.js
import Ember from 'ember';
import stateFor from 'path/to/state-for';

export default Ember.Component.extend({
  state: stateFor('email', { key: 'email' }),
});
```

**Comment by stefanpenner** [comment]
@thoov the same way CP's work. (it would just be a type of CP)

**Comment by thoov** [comment]
Ok let me whip something up

**Comment by stefanpenner** [comment]
:+1:

**Comment by jasonmit** [comment]
Nice work @thoov.  Docs need to be updated to reflect the PR :+1:

**Comment by thoov** [comment]
@jasonmit ya I am sending another PR with an updated example and I will be updating the readme
